### DISCUSSION TITLE
List specific value from datasource depending on env var

### DISCUSSION THREAD

**Discussion by v-v18** [hairyhenderson/gomplate#1634] [Q&A] [answered]
Hello,

I would like to render my template with a datasource but I would like to state the specific value from the datasource file as a variable (which is an env var). It would be good if I could solve the problem with only **one env.yml file** and only **one test.template file**.

So for example, I have the following files
_test.template (the syntax does not work but is to illustrate my problem)_
```
{{ (datasource "config").{{ .Env.ENVIRONMENT }}.value }}
```
and _env.yml_
```
region1:
    value: 123
region2:
    value: 456
```
If I set `export ENVIRONMENT=region1`, then I would like test.template to be rendered as `{{ (datasource "config").region1.value }}` which would give me `123`. Equally, with  `export ENVIRONMENT=region2` I would like to get `456`. So you can imagine that I have one pipeline for 2 environments and for each environment I would like to get a different rendered template.

Is there any way to do that? I had a look at [nested template](https://github.com/hairyhenderson/gomplate/discussions/1239) but that is not my use case as I want to variablize the map path and not the datasource file. Besides, I have only one **env.yml file**.

I would appreciate any help, thanks in advance.

<br> </br>
<br> </br>
### My attempts that did _not_ work
```
{{ $region := .Env.ENVIRONMENT }}
{{ (datasource "config").$region.value }}
```

This works when I do gomplate command twice, but I would be curious if there is a way to do it with just one gomplate command. Also, the syntax is difficult to maintain I believe.
```
{ "{{" }} (datasource "config").{{ .Env.ENVIRONMENT }}.value {{ "}}" }}
```

**Comment by hairyhenderson** [comment, marked as answer]
Hi @viet-ng95! I've converted this to a discussion since this is a usage question and not a bug report.

What you need here is the `index` function, and its usage for accessing maps (such as was provided by your datasource) is documented [here](https://docs.gomplate.ca/syntax/#maps).

What might work is something like:

```
{{ $config := datasource "config" -}}
{{ $region := index $config .Env.ENVIRONMENT -}}
{{ $region.value }}
```

Or, if you want to be less verbose and do it on a single line:

```
{{ (index (ds "config") .Env.ENVIRONMENT).value }}
```

Hope that helps!

**Reply by v-v18** [reply]
Thanks a bunch, that is perfect!
